src/sources.rs

```diff
@@ -0,0 +1,20 @@
+//! Sources of data we want to sync from.
+
+use anyhow::Result;
+use async_trait::async_trait;
+
+use crate::zitadel::SourceDiff;
+
+pub mod csv;
+pub mod ldap;
+pub mod ukt;
+
+/// A source of data we want to sync from.
+#[async_trait]
+pub trait Source {
+	/// Get source name for debugging.
+	fn get_name(&self) -> String;
```

Comment:
```rust
const NAME: &static str;
```
maybe?

Reply:
Done.

Reply:
I meant it doesn't have to be a trait method, it can be associated static value:
```rust 
pub trait Source {
  const NAME: &static str;
}

impl Source for SomeSource {
  const NAME: &static str = "Some source";
}

fn asdf(source: impl Source) {
  println!("{}", source::NAME);
}
```

Reply:
Yeah, but then it cannot be used as a trait object. So it won't be possible to gather the sources into a vector and iterate over them to sync each. And we would be back to where we were a couple of days ago.

Reply:
Huh, indeed